InstructorFeedbackEditPageUiTest failing on live server

java.lang.AssertionError: 
	at org.testng.AssertJUnit.fail(AssertJUnit.java:59)
	at org.testng.AssertJUnit.assertTrue(AssertJUnit.java:24)
	at org.testng.AssertJUnit.assertTrue(AssertJUnit.java:33)
	at teammates.test.cases.ui.browsertests.InstructorFeedbackEditPageUiTest.testAjaxOnVisibilityMessageButton(InstructorFeedbackEditPageUiTest.java:382)
	at teammates.test.cases.ui.browsertests.InstructorFeedbackEditPageUiTest.testGeneralQuestionOperations(InstructorFeedbackEditPageUiTest.java:108)
	at teammates.test.cases.ui.browsertests.InstructorFeedbackEditPageUiTest.allTests(InstructorFeedbackEditPageUiTest.java:82)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)

--- a/src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java
+++ b/src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java
@@ -399,10 +399,7 @@ public class InstructorFeedbackEditPageUiTest extends BaseUiTestCase {
         feedbackEditPage.changeQuestionTypeInForm(1, "InvalidQuestionType");
         feedbackEditPage.clickVisibilityPreviewForQuestion1();
         feedbackEditPage.waitForAjaxErrorOnVisibilityMessageButton(1);
-        
         assertFalse(feedbackEditPage.verifyVisibilityMessageIsDisplayed(1));
-        String errorMessage = "Visibility preview failed to load.";
-        assertTrue(feedbackEditPage.getPreviewLabel(1).getText().contains(errorMessage));
     }
 
     private void testDeleteQuestionAction(int qnNumber) {

--- a/src/test/java/teammates/test/pageobjects/AppPage.java
+++ b/src/test/java/teammates/test/pageobjects/AppPage.java
@@ -246,6 +246,16 @@ public abstract class AppPage {
     }
     
     /**
+     * Waits for text contained in the element to appear in the page, or timeout
+     * @param by
+     * @param text
+     */
+    public void waitForTextContainedInElementPresence(By by, String text) {
+        WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.inst().TEST_TIMEOUT);
+        wait.until(ExpectedConditions.textToBePresentInElementLocated(by, text));
+    }
+    
+    /**
      * Switches to the new browser window just opened.
      */
     protected void switchToNewWindow() {

--- a/src/test/java/teammates/test/pageobjects/AppPage.java
+++ b/src/test/java/teammates/test/pageobjects/AppPage.java
@@ -247,8 +247,6 @@ public abstract class AppPage {
     
     /**
      * Waits for text contained in the element to appear in the page, or timeout
-     * @param by
-     * @param text
      */
     public void waitForTextContainedInElementPresence(By by, String text) {
         WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.inst().TEST_TIMEOUT);

--- a/src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java
+++ b/src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java
@@ -838,8 +838,9 @@ public class InstructorFeedbackEditPage extends AppPage {
     }
     
     public void waitForAjaxErrorOnVisibilityMessageButton(int questionNumber) {
-        By buttonSelector = By.cssSelector("#visibilityMessageButton-" + questionNumber + " .active");
-        waitForElementToDisappear(buttonSelector);
+        String errorMessage = "Visibility preview failed to load.";
+        By buttonSelector = By.cssSelector("#visibilityMessageButton-" + questionNumber);
+        waitForTextContainedInElementPresence(buttonSelector, errorMessage);
     }
 
     public void clickResponseVisiblityCheckBoxForNewQuestion(String checkBoxValue) {

